src/components/organisms/EventModal/EventModal.tsx

```diff
@@ -63,12 +61,21 @@ export const EventModal: React.FC<EventModalProps> = ({
 
   const eventLocationToDisplay =
     (event.room || eventVenue?.name) ?? event.venueId;
-
   const goToEventLocation = () => {
     onHide();
 
     const { room = "" } = event;
     const roomUrlParam = getUrlParamFromString(room);
+    const [selectedRoom] =
+      eventVenue?.rooms?.filter(
+        (venueRoom) => getUrlParamFromString(venueRoom.title) === room
+      ) || [];
+
+    if (selectedRoom) {
+      openUrl(selectedRoom?.url);
+
+      return;
+    }
```

Comment:
Is it possible to just reuse `useRoom` here? It looks very hacky

Reply:
useRoom is already there ( 57th row ) , but it doesn't work for us in this case
we need to check both event and room instead of just room as done on map

Reply:
we should still use `enterRoom` in this case as it handles the case if the URL is not a venue

![image](https://user-images.githubusercontent.com/63194656/126805579-473218b2-d96e-4a71-86cc-44ea78fdceb1.png)

Reply:
```
const selectedRoom =
      eventVenue?.rooms?.find(
        (venueRoom) => getUrlParamFromString(venueRoom.title) === room
      );
```